### PR TITLE
feat: Simplify DLStreamer image to use a common image

### PR DESCRIPTION
**Pull request by brian-intel**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [x] Unit Tests have been added for new changes
- [x] Updated Documentation as relevant to the changes
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?

Consolidate the DLStreamer image and related code.

## Issue this PR will close

close: #289

## Test Instructions if applicable
1. make build-dlstreamer
2. ./camera-simulator/camera-simulator.sh
3. PIPELINE_PROFILE="gst" sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,15 +10,11 @@
 
 MKDOCS_IMAGE ?= asc-mkdocs
 
-build-all: build-soc build-dgpu
+build-dlstreamer:
```
Comment (jim-wang-intel) [inline Makefile:13]: fix the .PHONY

File: Makefile [hunk 2]
```diff
@@ -10,15 +10,11 @@
...
+	docker build --no-cache --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg HTTP_PROXY=${HTTP_PROXY} --target build-default -t dlstreamer:2.0 -f Dockerfile.dlstreamer .
 
-build-soc:
-	echo "Building for SOC (e.g. TGL/ADL/Xeon SP/etc) HTTPS_PROXY=${HTTPS_PROXY} HTTP_PROXY=${HTTP_PROXY}"
-	docker build --no-cache --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg HTTP_PROXY=${HTTP_PROXY} -t sco-soc:2.0 -f Dockerfile.soc .
-
-build-dgpu:
-	echo "Building for dgpu Arc/Flex HTTPS_PROXY=${HTTPS_PROXY} HTTP_PROXY=${HTTP_PROXY}"
-	docker build --no-cache --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg HTTP_PROXY=${HTTP_PROXY} -t sco-dgpu:2.0 -f Dockerfile.dgpu .
+build-dlstreamer-realsense:
```
Comment (jim-wang-intel) [inline Makefile:16]: same here- fix the .PHONY

File: Makefile [hunk 3]
```diff
@@ -10,15 +10,11 @@
 
 MKDOCS_IMAGE ?= asc-mkdocs
 
-build-all: build-soc build-dgpu
```
Comment (jim-wang-intel) [inline Makefile:13]: so build-all should build both dlstreamer without realsense and with realsense? what's the definition of build-all now

File: docs_src/pipelinesetup.md [hunk 4]
```diff
@@ -33,11 +33,9 @@ sh modelDownload.sh
...
 
-| Platform                                   | Docker Build Command       | Check Success                                     |
-| ------------------------------------------ | -------------------------- |---------------------------------------------------|
-| Intel platforms with Intel integrated GPUs | <pre>make build-soc</pre>  | docker images command to show <b>sco-soc:2.0</b>  |
-| Intel platforms with Intel discrete GPUs   | <pre>make build-dgpu</pre> | docker images command to show <b>sco-dgpu:2.0</b> |
-| build both platforms                       | <pre>make build-all</pre>  | docker images command to show both above          |
+| Platform                                   | Docker Build Command       | Check Success                                                |
+| ------------------------------------------ | -------------------------- |--------------------------------------------------------------|
+| Intel platforms using DLStreamer           | <pre>make build-dlstreamer</pre>  | docker images command to show <b>sdlstreamer:2.0</b>  |
```
Comment (jim-wang-intel) [inline docs_src/pipelinesetup.md:38]: should we also include the build-dlstreamer-realsense? it is worth documentation here too so for realsense users they can build with it

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (brian-intel) [reply Makefile:13]: Would it make sense to also add the OVMS docker builds and use this as a true build all

**Review by antoniomtz — APPROVED** [review]
LGTM

File: configs/opencv-ovms/scripts/run_gst.sh [hunk 5]
```diff
@@ -36,7 +36,7 @@ else
 fi
 
 echo $rmDocker
-TAG=sco-soc:2.0
+TAG=dlstreamer:2.0
```
Comment (jim-wang-intel) [inline configs/opencv-ovms/scripts/run_gst.sh:39]: sorry i missed this in my PR: this line can be removed, this env is no longer used.

**Review by vli11 — CHANGES_REQUESTED** [review]
the test instruction is for old dlstreamer way to run the pipeline, you may want to update the new test instruction for PR testing

**Review by vli11 — APPROVED** [review]
tested with `PIPELINE_PROFILE="gst" sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0`

**Pull request merged**